### DISCUSSION TITLE
Add an option or variable for automatically peer listening port "Open" state

### DISCUSSION THREAD

**Discussion by ghost** [transmission/transmission#6526] [Ideas]
Hi!

My Transmission works great with my VPN.

The problem is that the peer listening port after a reconnection or restart of Transmission often goes to "Closed" instead of "Open" and the sharing speed becomes much worse than when the peer listening port is "Open" .

The state between "Open" and "Closed" is completely random.

Would it be possible to add an option or variable so that Transmission restarts automatically until the peer listening port returns to "Open" state?

![295686527-9ccb65f6-69f2-4829-a29d-59bde69a9c50](https://github.com/transmission/transmission/assets/90903325/e6bf4314-5e95-45ea-8319-29bfce15f96f)

**Comment by tearfur** [comment]
I know some VPN allows you to configure port forwarding, you'd just to have to make sure you always have that configured for the VPN you are connecting to.
